### PR TITLE
CAD-792 mainnet-benchmark.sh:  work-limited, local chainsync benchmark

### PR DESCRIPTION
**Pull request by deepfire**
- this adds a locally-served chainsync benchmark script, with integrated profiling & whistles.
- this has been in intensive use over the past couple of weeks, in the course of monitoring performance investigations.

- This PR **does not result** in breaking changes to upstream dependencies.

Checklist
---------
- [x] This PR contains all the work required to resolve the linked issue.

- [ ] The work contained has sufficient documentation to describe what it does and how to do it.

- [x] The work has sufficient tests and/or testing.

- [x] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [x] I have added the appropriate labels to this PR.

### CODE REVIEW

File: scripts/mainnet-benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  Benchmark setup options:
+
+    --skip-prefetch             skip the prefetch phase
+    --prebench-pause            after prefetch, pause until 'Enter' is hit
+    --server-wait SECONDS       delay the benchmarked (client) side to start,
+                                  for this long -- to avoid profile disturbance.
+                                  Defaults to ${server_wait} seconds.
+    --no-server-wait            same as --server-wait 0
+    --node-stdout FILE          redirect stdout of benchmarked node to a file
+    --node-no-stdout            redirect stdout of benchmarked node to Dave Null
```
Comment (ArturWieczorek) [inline scripts/mainnet-benchmark.sh:30]: You meant Dev Null ?

Comment (erikd) [inline scripts/mainnet-benchmark.sh:30]: That is cute and all, but I really think it should say `/dev/null` to reduce any chance of a misunderstanding.

Comment (intricate) [reply scripts/mainnet-benchmark.sh:30]: Agree with @erikd, we should change this to `/dev/null`.

**Comment by deepfire** [comment]
> I would prefer something like this, that lets you easily connect to an external server.

I see -- I'll add this functionality in my spare time, as this work is not currently planned in our sprint.

File: scripts/mainnet-benchmark.sh (hunk 1 above)
Comment (deepfire) [reply scripts/mainnet-benchmark.sh:30]: I was pretty sure this will attract attention : -)

Pushed.

**Comment by deepfire** [comment]
Changing status to WIP, to reflect necessity of feature change, as suggested by Karl.

**Comment by CodiePP** [comment]
outdated? what is the status of this PR?

**Comment by deepfire** [comment]
No, not outdated, likely to be resurrected.

**Comment by johnalotoski** [comment]
Remove the `hydra-dont-build` label once work resumes on the PR.  This will reduce eval load on hydra until then.  Thanks.

**Comment by Jimbo4350** [comment]
Closing this. If this is still relevant please re-open a new PR.

**Pull request closed without merging**